### PR TITLE
[RKE2] Cluster unable to provision after switching back to Back to Form from Edit as YAML

### LINKED ISSUE #8606: 'View Yaml' action from list views fails for users who can list but not get a resource
To reproduce:
1. Create a custom cluster to role to list a resource
2. Create a standard user
3. Explore a cluster, navigate to 'Cluster & Project Members'
4. Add the standard user to the cluster with the custom role created in step 1
5. Login as the standard user and navigate to the relevant resource's list view
6. Use the kebab menu to select 'view as yaml' for one of the resources listed
Result: The yaml doesn't load and an error is seen in the console:
```
console.js:31 TypeError: Cannot read properties of null (reading 'doneOverride')
    at f.<anonymous> (index.vue?46a7:1:1374)
    at t._render (vue.runtime.esm.js:3569:22)
    at f.r (vue.runtime.esm.js:4081:21)
    at Tn.get (vue.runtime.esm.js:4495:25)
    at Tn.run (vue.runtime.esm.js:4570:22)
    at Sn (vue.runtime.esm.js:4326:13)
    at Array.<anonymous> (vue.runtime.esm.js:1989:12)
    at ce (vue.runtime.esm.js:1915:12)
```
Expected result:
The user can view the yaml of the resource

It's worth noting that 'view config' still works in this scenario. This is probably because when we go from a list view to 'view config' we use data already loaded from the request to find all of that resource, but when we 'view yaml' we re-query the api for that specific resource in yaml form.

### LINKED ISSUE #9248: [RKE2] Cluster unable to provision after switching back to `Back to Form` from `Edit as YAML`
**Setup**
- Rancher version: v2.7-head c402790
- Rancher UI Extensions:
- Browser type & version: Chrome Build 114

**Describe the bug**

When creating a cluster, I went to the `Edit as YAML` and went back to the form edit, I tried to create the cluster but it failed with an error saying `"undefined" is not valid in JSON`
image

**To Reproduce**

1. Provision RKE2 AWS node driver cluster on v2.7-head
2. Fill out the default values for provisioning the cluster
3. Click `Edit as YAML`
4. Click `Back to Form`
5. Click `Create`

**Result**
Rancher hits a white page with the error `"undefined" is not valid in JSON`

**Expected Result**

Rancher should be able to provision the cluster since all the default values to create the cluster are filled out. When I do the same steps without going to the YAML and back to form, it works as expected.

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #9248

### Occurred changes and/or fixed issues
While toggling YAML
- Application is not breaking anymore 
- Data for the Agent configuration is not removed anymore (it was not reported)

### Technical notes summary
- Fixed issue with before hook called while toggling the YAML and resetting data, it's theoretically called for saving resources but it's used also for this case looks like
- Moved function to usage component due single use case
- Moved condition over prop to parent since is initialized
- Added unit tests
- Added directive for `clean-tooltip`

### Screenshot/Video

https://github.com/rancher/dashboard/assets/5009481/79bef742-83a4-4597-998e-c574148442ea

### CODE REVIEW

**Comment by cnotv** [comment]
@aalves08 Although the issue is solved, I see (placing as content in the UI) that the data is removed. I imagine that this is an issue for the agent. Should we maybe handle it in somehow?


https://github.com/rancher/dashboard/assets/5009481/fd16e7f9-a351-4b9e-9459-3576f560fd81

**Comment by cnotv** [comment]
Due to our setup and architecture it's troublesome to test the async hook `fetch()` left out from Nuxt migration, as it's not part of the `vue/test-util`. It would be not possible then without add any extra setup to test it.

Here's some ways async or `fetch()` are tackled down:
- [Old test-util for async functions](https://v1.test-utils.vuejs.org/guides/testing-async-components.html)
- [New version](https://test-utils.vuejs.org/guide/advanced/async-suspense.html#resolving-other-asynchronous-behavior)
- [Nuxt integration into tests (2019 open issue)](https://github.com/testing-library/vue-testing-library/issues/92)
- [Nuxt 3 test library](https://nuxt.com/docs/getting-started/testing#fetchurl-1)


None of these cases seems working for us.

**Comment by cnotv** [comment]
Looks like finally this was the cause of the issue in the test:
```ts
      const extClass = this.$plugin.getDynamic('provisioner', this.provider);

      if (extClass) {
        return new extClass({
          dispatch: this.$store.dispatch,
          getters:  this.$store.getters,
          axios:    this.$store.$axios,
          $plugin:  this.$store.app.$plugin,
          $t:       this.t,
          isCreate: this.isCreate
        });
      }
```

File: shell/edit/provisioning.cattle.io.cluster/AgentConfiguration.vue [hunk 1]
```diff
@@ -215,7 +186,7 @@ export default {
 </script>
 
 <template>
-  <div v-if="value && Object.keys(value).length">
```
Comment (aalves08) [inline shell/edit/provisioning.cattle.io.cluster/AgentConfiguration.vue:218]: Why did you remove this condition? I am sure it was added with a specific purpose...

File: shell/edit/provisioning.cattle.io.cluster/rke2.vue [hunk 2]
```diff
@@ -1615,8 +1663,8 @@ export default {
 
       // store the current data for fleet and cluster agent so that we can re-apply it later if the save fails
       // we also have a before hook (check created() hooks) where the cleanup of the data occurs
```
Comment (aalves08) [inline shell/edit/provisioning.cattle.io.cluster/rke2.vue:1665]: this comment could be removed with your changes, I believe

**Review by aalves08 — COMMENTED** [review]
left a couple of comments
